### DISCUSSION TITLE
Question:

### DISCUSSION THREAD

**Discussion by hlkavya1311** [InboxSDK/InboxSDK#929] [General]
Hi team,

We have a Chrome extension [https://chrome.google.com/webstore/detail/fyle-expense-management/abggpefphmldapcoknbcaadbpdjjmjgk ](https://chrome.google.com/webstore/detail/fyle-expense-management/abggpefphmldapcoknbcaadbpdjjmjgk)  which is used to submit business expenses.
We were able to get the file attachments from the emails earlier using the `getDownloadURL() `method since it's deprecated we can no longer get the attachments without user interaction. 
We figure out a way to do this using the `_attachmentCardImplementation._element` of AttachmentCardClickEvent options. 
Can you please confirm if this is okay to use the `_element` options to get the download URLs?
**Also is there any chance` getDownloadURL() `will be back in the later updates?**

Thanks,
Kavya H L

**Comment by Macil** [comment]
We originally deprecated the `getDownloadURL()` method because in Google Inbox it wasn't possible for us to get the download URL without user interaction and we didn't want to support APIs that wouldn't work in Inbox too, but given that Google Inbox is gone and it looks like it's still possible to get the URL without user interaction in Gmail, there's little reason for us to not bring back `getDownloadURL()`. We can bring it back.

We don't recommend depending on any undocumented underscore-prefixed properties as they weren't intentionally exposed and incompatible changes may happen in them.

**Comment by hlkavya1311** [comment]
Thank you for the reply @Macil, this was really helpful. Is there any specific timeline or ETA to bring back the `getDownloadURL()`? This would really help us further to make any product decisions.

Thanks,
Kavya H L

**Comment by abhishek1234321** [comment]
@Macil If we were to give a PR for this and if it gets approved, would we be able to fast-track and merge it in?

**Comment by Macil** [comment]
Yes, we would be open to taking a PR for this.
